hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/components/HawkbitUIErrorHandler.java

```diff
@@ -38,10 +42,21 @@ public void error(final ErrorEvent event) {
 
         final Optional<Page> originError = getPageOriginError(event);
 
+        final HawkbitErrorNotificationMessage message = buildNotification(getRootExceptionFrom(event));
+
         if (originError.isPresent()) {
-            final HawkbitErrorNotificationMessage message = buildNotification(getRootExceptionFrom(event));
+            final Connector connector = ((ConnectorErrorEvent) event).getConnector();
+            // in case of BulkUpload: BulkUpload needs a lock to show the
+            // notification
+            if (connector instanceof UI) {
+                ((UI) (((ConnectorErrorEvent) event).getConnector())).access(() -> message.show(originError.get()));
```

Comment:
I don't get the difference between line 52 and 55. Where is the mentioned lock? You are showing the message anyway !?!?